lib/linklist.h

```diff
@@ -25,6 +25,11 @@
 extern "C" {
 #endif
 
+/* Typedef of call back function passed to list_cond_delete_nodes
+ * where this function decides if a node has to be deleted
+ */
+typedef bool (*cond_del_fn_t)(void *data);
```

Comment:
No new typedefs in code without a very good justification. Please remove this.

Reply:
will remove it